generators/server/templates/_pom.xml

```diff
@@ -974,9 +977,6 @@
                     Enable the line below to have remote debugging of your application on port 5005
                     <jvmArguments>-agentlib:jdwp=transport=dt_socket,server=y,suspend=n,address=5005</jvmArguments>
                     -->
-                    <arguments>
```

Comment:
I found the root cause for this. The problem is that when we set a `
spring.profiles.default` spring boot loads up `application.yml` and
`application.dev.yml`, when we set `spring.profiles.active`to prod spring
boot loads up `application.yml`, `application.prod.yml` and
`application.dev.yml` and hence some of the stuff like DB ends up reading
from the dev properties.

I'm trying to find a fix using a listner with
ApplicationListener<ApplicationEvent>

Ill post my findings

Thanks & Regards,
Deepu

On Fri, May 20, 2016 at 4:31 PM, Erik Kemperman <notifications@github.com>
wrote:

> In generators/server/templates/_pom.xml
> <https://github.com/jhipster/generator-jhipster/pull/3620#discussion_r64006529>
> :
>
> > @@ -974,9 +977,6 @@
> >                      Enable the line below to have remote debugging of your application on port 5005
> >                      <jvmArguments>-agentlib:jdwp=transport=dt_socket,server=y,suspend=n,address=5005</jvmArguments>
> >                      -->
> > -                    <arguments>
>
> @deepu105 <https://github.com/deepu105> I am finding the same thing. I
> didn't notice earlier, because we use H2 for production as well.
>
> You are right that adding
>
> <arguments>
>     <argument>spring.profiles.active=${spring.profiles.active}</argument>
> </arguments>
>
> to the spring-boot plugin configuration "works". But it seems to me that
> this is just suppressing the symptom of a deeper problem.
>
> It looks to me like this is a chicken/egg situation around the order in
> which:
>
> -- Spring looks to VM arguments for explicit profiles set to active
> -- Database config looks for properties from active profiles
> -- Spring looks at application.yml for active profiles
>
> I saw similar weirdness when working on the original default profile
> stuff, which was why my original effort ended up using a static initializer
> block (in BuildProperties.java) to make sure that the default profile was
> set before anything else happened.
>
> I am curious if this is because the DB configuration uses annotations like
> @ConfigurationProperties and @Profile. Would it maybe help to just check
> for profile dynamically instead?
>
> —
> You are receiving this because you were mentioned.
> Reply to this email directly or view it on GitHub
> <https://github.com/jhipster/generator-jhipster/pull/3620/files/bae620709f7353d9bfb1e288e6159fc7acda9d3b#r64006529>
>